### DISCUSSION TITLE
How to get the lineal_path_distribution result by porespy

### DISCUSSION THREAD

**Discussion by dingyineversad** [PMEAL/porespy#471] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/77733099/114327381-5f715180-9b6b-11eb-90a7-53932087459f.png)
Sorry to interrupt again, could you please tell me how to get the lineal_path_distribution result by porespy?
Thank you very much,professor

**Comment by jgostick** [comment, marked as answer]
Hi @dingyineversad, I have cooked up an example in the form of a Jupyter Notebook and added it to our ``new_examples`` branch.  You can see it [here](https://porespy.org/examples/metrics/tutorials/lineal_path_function.html).

**Reply by dingyineversad** [reply]
![image](https://user-images.githubusercontent.com/77733099/114377703-89e9fb80-9bb9-11eb-8b27-441be061c744.png)
When I run this line of code ,it shows this

**Reply by jgostick** [reply]
Ahhh, I added the linearl_path_function recently, so you probably don't have the latest version.  You can see the source code [here](https://github.com/PMEAL/porespy/blob/c1e623ad73b5814c4d937260e64cacf82c38dd6e/porespy/metrics/__funcs__.py#L206).  It's very short, so you can just copy-paste it into your script.  You can remove all the docstring text at the top to shorten it up.  You'll need to add ``from collections import namedtuple`` and you'll also need [_parse_histogram function here](https://github.com/PMEAL/porespy/blob/c1e623ad73b5814c4d937260e64cacf82c38dd6e/porespy/metrics/__funcs__.py#L556).

**Comment by dingyineversad** [comment]
OK, I see. Thank you very much
